service/storage_proxy.cc

```diff
@@ -4787,9 +4795,9 @@ void storage_proxy::init_messaging_service() {
         utils::UUID schema_version = in.schema_version();
         return handle_write(src_addr, t, schema_version, std::move(in), std::move(forward), reply_to, shard, response_id,
                 trace_info ? *trace_info : std::nullopt,
-                /* apply_fn */ [] (shared_ptr<storage_proxy>& p, tracing::trace_state_ptr, schema_ptr s, const frozen_mutation& m,
+                /* apply_fn */ [] (shared_ptr<storage_proxy>& p, tracing::trace_state_ptr tr_state, schema_ptr s, const frozen_mutation& m,
                         clock_type::time_point timeout) {
-                    return p->mutate_locally(std::move(s), m, db::commitlog::force_sync::no, timeout);
+                    return p->mutate_locally(std::move(s), m, std::move(tr_state), db::commitlog::force_sync::no, timeout);
```

Comment:
Note sure if std::move() is legal here, please drill down.

Reply:
it should be ok - it only moves a local function parameter, which was passed by value

Reply:
The question is if the caller calls the function multiple times.

Reply:
Hm, I don't follow. If the signature of this function is 
```cpp
(shared_ptr<storage_proxy>& p, tracing::trace_state_ptr tr_state, schema_ptr s, const frozen_mutation& m, clock_type::time_point timeout)
```
then `tr_state` is taken by value, so whatever the caller passed here will be copied, right? Or did I miss something.

Reply:
@psarna Please, note that `trace_state_ptr` is not initialized anywhere, nor it's used (in the master).
@gleb-cloudius It seems that you've added it. Looks like some artifact that is left by mistake.